lib/Elevate/Blockers/Databases.pm

```diff
@@ -89,28 +91,15 @@ sub _blocker_acknowledge_postgresql_datadir ($self) {
 
 sub _has_mapped_postgresql_dbs ($self) {
 
-    # Normalize locale:
-    local $ENV{HTTP_COOKIE} = 'session_locale=en';
-
-    # Get a list of cPanel users:
-    my $users_json   = Cpanel::SafeRun::Simple::saferunnoerror(qw(/usr/local/cpanel/bin/whmapi1 --output=json list_users));
-    my $users_parsed = Cpanel::JSON::Load($users_json);
-    die "WHM API1 list_users call returned result 0: $users_parsed->{metadata}->{reason}" unless $users_parsed->{metadata}->{result};
-    my @users = grep { $_ ne "root" } $users_parsed->{data}->{users}->@*;
-
-    # Compile a list of users with PgSQL DBs:
-    my @users_with_dbs;
-    foreach my $user (@users) {
-        my $dbs_json   = Cpanel::SafeRun::Simple::saferunnoerror( "/usr/local/cpanel/bin/uapi", "--user=$user", qw(--output=json Postgresql list_databases) );
-        my $dbs_parsed = Cpanel::JSON::Load($dbs_json);
-        if ( !$dbs_parsed->{result}->{status} ) {
-            return if ( $dbs_parsed->{result}->{errors}->[0] // '' ) eq 'This server does not support this functionality.';    # sanity check
-            die( "UAPI Postgresql::list_databases call returned status 0 for user $user:\n" . join( "\n", $dbs_parsed->{result}->{errors}->@* ) );
-        }
-        push @users_with_dbs, $user if scalar $dbs_parsed->{result}->{data}->@*;
+    my $dbindex = eval { Cpanel::DB::Map::Collection::Index->new( { db => 'PGSQL' } ); };
+    if ( my $exception = $@ ) {
+        ERROR( 'Unable to read the database index file:  ' . Cpanel::Exception::get_string($exception) );
+        return ();
```

Comment:
We should probably block if we hit this exception.  This would also be an indication that the mysql reinstall is going to go very poorly since the mysql user map is contained in this file too, correct?

Reply:
It'd get rebuilt when /ulc/bin/dbindex gets run.

Reply:
Yeah, there should be a blocker for this.  It is an unusual situation; but, it also would mean something is really awry.